Overhaul versioning doc (#381)
vegaprotocol/documentation: VERSIONING.md

- _/docs/mainnet/_ represents the last deployed version on [mainnet](https://blog.vega.xyz/what-to-expect-from-restricted-mainnet-616086d9fdaf)
- _/docs/testnet/_ represents the most recent version on [Fairground testnet](https://fairground.wtf)

# Overview
![update-process png excalidraw](https://user-images.githubusercontent.com/6678/200811843-564a0127-2f25-4f6b-ae56-10305cf1ddfa.png)

Every time a release is [tagged in vegaprotocol/vega](https://github.com/vegaprotocol/vega), a [pull request is created in this repository](https://github.com/vegaprotocol/documentation/pulls?q=is%3Apr+automated+specs+update).  

1. When the release is adopted on testnet, in that branch, follow the [Changing the testnet version](#changing-the-testnet-version) instructions
2. Patch releases that are deployed to the network follow the same update process as above. Patches that do not get deployed to testnet don't require an update - the pull request can be closed.
3. When a release is adopted on mainnet, follow the [Changing the mainnet version](#changing-the-mainnet-version) instructions.

--- 

## tl;dr

> - [/docs/](https://github.com/vegaprotocol/documentation/tree/main/versioned_docs/) is 'the current testnet version'. Every time testnet is updated, the docs here will be updated. This folder is served as `/testnet/`.

Reviewer: this URL doesn't look right

> - [/docs/](https://github.com/vegaprotocol/documentation/tree/main/versioned_docs/) is 'the current testnet version'. Every time testnet is updated, the docs here will be updated. This folder is served as `/testnet/`.

Reviewer: ```suggestion
- [/docs/](https://github.com/vegaprotocol/documentation/tree/main/docs/) is 'the current testnet version'. Every time testnet is updated, the docs here will be updated. This folder is served as `/testnet/`.
```

> - [/docs/](https://github.com/vegaprotocol/documentation/tree/main/versioned_docs/) is 'the current testnet version'. Every time testnet is updated, the docs here will be updated. This folder is served as `/testnet/`.

Author: ```suggestion
- [/docs/](https://github.com/vegaprotocol/documentation/tree/main/docs/) is 'the current testnet version'. Every time testnet is updated, the docs here will be updated. This folder is served as `/testnet/`.
```

> - [/versioned_docs/](https://github.com/vegaprotocol/documentation/tree/main/versioned_docs/) contains *every version that has been released to mainnet*. The most recent version in this folder is served as `/mainnet/`.

Reviewer: Doesn't this sort of contradict the line below about how only two mainnet versions of docs are kept?

> - [/versioned_docs/](https://github.com/vegaprotocol/documentation/tree/main/versioned_docs/) contains *every version that has been released to mainnet*. The most recent version in this folder is served as `/mainnet/`.

Author: ```suggestion
- [/versioned_docs/](https://github.com/vegaprotocol/documentation/tree/main/versioned_docs/) contains version folders for versions that ran acceptably on testnet and were release candidates for mainnet. The current strategy is to only keep the most recent version in this folder, which is served as `/mainnet/`.
```

> - Only two mainnet versions are kept. [vegaprotocol/vega](https://github.com/vegaprotocol/vega)versions that are not deployed to testnet are not built. 

Reviewer: Should this say "deployed to mainnet"? I wonder if it should also be more explicit, since versions that go to mainnet are a culmination of all the software versions, just not all used for mainnet individually

Author: I have made this made more sense

## Changing the testnet version

> When a new release is *tagged*, a pull request with the API documents will automatically be create. It will add a new folder in `/specs/`, for example `v0.88.1`. Only those versions that actually get released need to be merged, so **when the network is updated**:

Reviewer: ```suggestion
When a new release is *tagged*, a pull request with the API documents will automatically be created. It will add a new folder in `/specs/`, for example `v0.88.1`. Only those versions that actually get released need to be merged, so **when the testnet network is updated**:
```

> When a new release is *tagged*, a pull request with the API documents will automatically be create. It will add a new folder in `/specs/`, for example `v0.88.1`. Only those versions that actually get released need to be merged, so **when the network is updated**:

Reviewer: ```suggestion
When a new release is *tagged*, a pull request with the API documents will automatically be created. It will add a new folder in `/specs/`, for example `v0.88.1`. Only those versions that actually get released need to be merged, so **when the network is updated**:
```
